packages/calypso-analytics/src/tracks.ts

```diff
@@ -27,21 +37,20 @@ const EVENT_NAME_EXCEPTIONS = [
 	'wcadmin_storeprofiler_payment_login',
 	'wcadmin_storeprofiler_payment_create_account',
 ];
-let _superProps; // Added to all Tracks events.
+let _superProps: any; // Added to all Tracks events.
 let _loadTracksResult = Promise.resolve(); // default value for non-BOM environments.
 
-if ( typeof window !== 'undefined' ) {
-	window._tkq = window._tkq || [];
-}
-
 if ( typeof document !== 'undefined' ) {
 	_loadTracksResult = loadScript( '//stats.wp.com/w.js?60' );
 }
 
-function createRandomId( randomBytesLength = 9 ) {
+function createRandomId( randomBytesLength = 9 ): string {
+	if ( typeof window === 'undefined' ) {
+		return '';
+	}
```

Comment:
If `window` isn't defined maybe this should go down the same code path as when `window.crypto` isn't defined.

Reply:
ahh I can't do that because in the case where `window.crypto` is undefined, it still uses the `window.btoa` function on line `:62`

Reply:
lol so it does, missed that 😂